examples/cpp/Breakpoint.cc

```diff
@@ -0,0 +1,156 @@
+/*
+ * This is an example of a hardware breakpoint on a kernel address.
+ * run in project examples directory with:
+ * sudo ./breakpoint.py <0xaddress> <pid> <breakpoint_type>
+ * HW_BREAKPOINT_W = 2
+ * HW_BREAKPOINT_RW = 3
```

Comment:
The above two lines are not needed in header comments. You can refer to a header file.

Reply:
> Thanks for working on this. Long term, we want to have a tool for people to watch symbols. But API support and examples with pid/address as an input is a great start.
> 
> @danobi has a prototype for bpftrace [iovisor/bpftrace#790](https://github.com/iovisor/bpftrace/pull/790). please take a look.
> Beyond inlined comments, a few general opinion below:
> . we should change name from breakpoint to watchpoint, which better reflects what bpf program tries to do.
> . it will be good if you can give more in-depth example with different bp_type's.
> . some indentation error and some lines with trailing spaces. you can check with `git show`.
> 
> Thanks a lot for the valuable insights on my work! I'll do the rest as you said. But please could you precisely explain what needs to be done in this regard?
 "it will be good if you can give more in-depth example with different bp_type's."